package de.oliver.fancynpcs.v26_4.attributes;

import de.oliver.fancynpcs.api.Npc;
import de.oliver.fancynpcs.api.NpcAttribute;
import net.minecraft.world.entity.raid.Raider;
import org.bukkit.entity.EntityType;
import org.bukkit.entity.Illager;

import java.util.ArrayList;
import java.util.Arrays;
import java.util.List;

public class IllagerAttributes {

    public static List<NpcAttribute> getAllAttributes() {
        List<NpcAttribute> attributes = new ArrayList<>();

        attributes.add(new NpcAttribute(
                "celebrating",
                List.of("true", "false"),
                Arrays.stream(EntityType.values())
                        .filter(type -> type.getEntityClass() != null && Illager.class.isAssignableFrom(type.getEntityClass()))
                        .toList(),
                IllagerAttributes::setCelebrating
        ));

        return attributes;
    }

    private static void setCelebrating(Npc npc, String value) {
        Raider raider = npc.getNmsEntity();

        boolean isCelebrating = Boolean.parseBoolean(value);

        raider.setCelebrating(isCelebrating);
    }

}
